src/systemTest/java/soot/asm/AsmMethodSourceOrigNamesTest.java

```diff
@@ -0,0 +1,1263 @@
+    opts.setPhaseOption("jb.sils", "enabled:false");
+  }
+
+  /**
+   * Below is the output of "javap -verbose" for the relevant method. The '|' character was added to mark the ranges where
+   * the LocalVariableTable specifies a variable name for local slot 9. Notice there is an additional use of slot 9 at offset
+   * 126 which does not appear in any of the specified ranges which means the local variable name for that use is unknown.
+   * 
+   * <code><pre>
+   *  public void write(char[], int, int) throws java.io.IOException;
+   *    descriptor: ([CII)V
+   *    flags: ACC_PUBLIC
+   *    Code:
+   *      stack=5, locals=10, args_size=4
+   *         0: iconst_3
+   *         1: iload_3
+   *         2: imul
+   *         3: istore        4
+   *         5: iload         4
+   *         7: sipush        16384
+   *        10: aload_0
+   *        11: getfield      #5                  // Field count:I
+   *        14: isub
+   *        15: if_icmplt     96
+   *        18: aload_0
+   *        19: invokevirtual #6                  // Method flushBuffer:()V
+   *        22: iload         4
+   *        24: sipush        16384
+   *        27: if_icmplt     96
+   *        30: iconst_1
+   *        31: iload_3
+   *        32: sipush        5461
+   *        35: idiv
+   *        36: iadd
+   *        37: istore        5
+   *        39: iconst_0
+   *        40: istore        6
+   *        42: goto          88
+   *        45: iload_2
+   *        46: iload_3
+   *        47: iload         6
+   *        49: imul
+   *        50: iload         5
+   *        52: idiv
+   *        53: iadd
+   *        54: istore        7
+   *        56: iload_2
+   *        57: iload_3
+   *        58: iload         6
+   *        60: iconst_1
+   *        61: iadd
+   *        62: imul
+   *        63: iload         5
+   *        65: idiv
+   *        66: iadd
+   *        67: istore        8
+   *        69: iload         8
+   *        71: iload         7
+   *        73: isub
+   *    |   74: istore        9
+   *    |   76: aload_0
+   *    |   77: aload_1
+   *    |   78: iload         7
+   *    |   80: iload         9
+   *    |   82: invokevirtual #7                  // Method write:([CII)V
+   *    |   85: iinc          6, 1
+   *        88: iload         6
+   *        90: iload         5
+   *        92: if_icmplt     45
+   *        95: return
+   *        96: iload_3
+   *        97: iload_2
+   *        98: iadd
+   *        99: istore        5
+   *       101: aload_0
+   *       102: getfield      #3                  // Field m_outputBytes:[B
+   *       105: astore        6
+   *       107: aload_0
+   *       108: getfield      #5                  // Field count:I
+   *       111: istore        7
+   *       113: iload_2
+   *       114: istore        8
+   *       116: goto          133
+   *       119: aload         6
+   *       121: iload         7
+   *       123: iinc          7, 1
+   *    ?  126: iload         9
```

Comment:
@tim-hoffman  Thanks for adding this test! t is a really strange the case. Even for other parts of slot 8, the starting offset in the table is not exactly as where the variable is used.

Reply:
Yes, from what I can tell, the starting offset is always the instruction following the store to the slot. I'm not sure why it is this way but it is at least consistent and I have addressed that: https://github.com/soot-oss/soot/blob/c295e358ec509d7ae93cc612a7de5e22c121e8ce/src/main/java/soot/asm/AsmMethodSource.java#L2325-L2327